### ISSUE TITLE
 crashed in append_formatv(std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >&, wchar_t const*, __va_list_tag*)()

### ISSUE THREAD

**Issue by zd5043039119** [fish-shell/fish-shell#5401]
Fish Version
```
fish,version 2.7.1
```
OS
```
Linux localhost.localdomain 4.19.2-300.fc29.x86_64 #1 SMP Wed Nov 14 19:05:24 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```
TERM
```
xterm
```

When i typed "fish" in shell after i update the system to latest by 'sudo dnf update -y', i got 
```
fish: 'fish' terminated by signal SIGSEGV (Address boundary error)
```
also the dump here
```
12月 11 14:58:56 localhost.localdomain systemd-coredump[32054]: Process 31751 (fish) of user 1000 dumped core.
                                                                 
                                                                 Stack trace of thread 31751:
                                                                 #0  0x00007f799f4198f5 __wcslen_avx2 (libc.so.6)
                                                                 #1  0x00007f799f315315 vfwprintf (libc.so.6)
                                                                 #2  0x00007f799f3c3e18 __vswprintf_chk (libc.so.6)
                                                                 #3  0x000055be89cb9f38 _Z14append_formatvRNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEPKwP13__va_list_tag (fish)
                                                                 #4  0x000055be89cbab88 _Z13append_formatRNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEPKwz (fish)
                                                                 #5  0x000055be89d22c1e _ZNK8parser_t20stack_trace_internalEmPNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEE.part.105 (fish)
                                                                 #6  0x000055be89d22e37 _ZNK8parser_t20stack_trace_internalEmPNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEE.part.105 (fish)
                                                                 #7  0x000055be89d22e37 _ZNK8parser_t20stack_trace_internalEmPNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEE.part.105 (fish)
                                                                 #8  0x000055be89d232b7 _ZN8parser_t12current_lineB5cxx11Ev (fish)
                                                                 #9  0x000055be89c8a3eb _Z18builtin_print_helpR8parser_tR12io_streams_tPKwR15output_stream_t (fish)
                                                                 #10 0x000055be89ca80b5 _Z11builtin_setR8parser_tR12io_streams_tPPw (fish)
                                                                 #11 0x000055be89c8af56 _Z11builtin_runR8parser_tPKPKwR12io_streams_t (fish)
                                                                 #12 0x000055be89cdb3d0 _Z8exec_jobR8parser_tP5job_t (fish)
                                                                 #13 0x000055be89d1413e _ZN25parse_execution_context_t9run_1_jobERK12parse_node_tPK7block_t (fish)
                                                                 #14 0x000055be89d1473e _ZN25parse_execution_context_t12run_job_listERK12parse_node_tPK7block_t (fish)
                                                                 #15 0x000055be89d14ab3 _ZN25parse_execution_context_t17run_for_statementERK12parse_node_tS2_ (fish)
                                                                 #16 0x000055be89d155f3 _ZN25parse_execution_context_t19run_block_statementERK12parse_node_t (fish)
                                                                 #17 0x000055be89d14443 _ZN25parse_execution_context_t9run_1_jobERK12parse_node_tPK7block_t (fish)
                                                                 #18 0x000055be89d1473e _ZN25parse_execution_context_t12run_job_listERK12parse_node_tPK7block_t (fish)
                                                                 #19 0x000055be89d15d60 _ZN25parse_execution_context_t19eval_node_at_offsetEjPK7block_tRK10io_chain_t (fish)
                                                                 #20 0x000055be89d221e2 _ZN8parser_t15eval_block_nodeEjRK10io_chain_t12block_type_t (fish)
                                                                 #21 0x000055be89d2249f _ZN8parser_t4evalERKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEERK10io_chain_t12block_type_t17parse_node_tree_t (fish)
                                                                 #22 0x000055be89d23ddd _ZN8parser_t4evalERKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEERK10io_chain_t12block_type_t (fish)
                                                                 #23 0x000055be89cdac8f _ZL20internal_exec_helperR8parser_tRKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEj12block_type_tRK10io_chain_t.constprop.157 (fish)
                                                                 #24 0x000055be89cdc49d _Z8exec_jobR8parser_tP5job_t (fish)
                                                                 #25 0x000055be89d1413e _ZN25parse_execution_context_t9run_1_jobERK12parse_node_tPK7block_t (fish)
                                                                 #26 0x000055be89d1473e _ZN25parse_execution_context_t12run_job_listERK12parse_node_tPK7block_t (fish)
                                                                 #27 0x000055be89d15d60 _ZN25parse_execution_context_t19eval_node_at_offsetEjPK7block_tRK10io_chain_t (fish)
                                                                 #28 0x000055be89d221e2 _ZN8parser_t15eval_block_nodeEjRK10io_chain_t12block_type_t (fish)
                                                                 #29 0x000055be89d2249f _ZN8parser_t4evalERKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEERK10io_chain_t12block_type_t17parse_node_tree_t (fish)
                                                                 #30 0x000055be89d23ddd _ZN8parser_t4evalERKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEERK10io_chain_t12block_type_t (fish)
                                                                 #31 0x000055be89cdac8f _ZL20internal_exec_helperR8parser_tRKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEj12block_type_tRK10io_chain_t.constprop.157 (fish)
                                                                 #32 0x000055be89cdc49d _Z8exec_jobR8parser_tP5job_t (fish)
                                                                 #33 0x000055be89d1413e _ZN25parse_execution_context_t9run_1_jobERK12parse_node_tPK7block_t (fish)
                                                                 #34 0x000055be89d1473e _ZN25parse_execution_context_t12run_job_listERK12parse_node_tPK7block_t (fish)
                                                                 #35 0x000055be89d152fa _ZN25parse_execution_context_t16run_if_statementERK12parse_node_t (fish)
                                                                 #36 0x000055be89d1442b _ZN25parse_execution_context_t9run_1_jobERK12parse_node_tPK7block_t (fish)
                                                                 #37 0x000055be89d1473e _ZN25parse_execution_context_t12run_job_listERK12parse_node_tPK7block_t (fish)
                                                                 #38 0x000055be89d15d60 _ZN25parse_execution_context_t19eval_node_at_offsetEjPK7block_tRK10io_chain_t (fish)
                                                                 #39 0x000055be89d221e2 _ZN8parser_t15eval_block_nodeEjRK10io_chain_t12block_type_t (fish)
                                                                 #40 0x000055be89d2249f _ZN8parser_t4evalERKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEERK10io_chain_t12block_type_t17parse_node_tree_t (fish)
                                                                 #41 0x000055be89d23ddd _ZN8parser_t4evalERKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEERK10io_chain_t12block_type_t (fish)
                                                                 #42 0x000055be89cdac8f _ZL20internal_exec_helperR8parser_tRKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEj12block_type_tRK10io_chain_t.constprop.157 (fish)
                                                                 #43 0x000055be89cdc49d _Z8exec_jobR8parser_tP5job_t (fish)
                                                                 #44 0x000055be89d1413e _ZN25parse_execution_context_t9run_1_jobERK12parse_node_tPK7block_t (fish)
                                                                 #45 0x000055be89d1473e _ZN25parse_execution_context_t12run_job_listERK12parse_node_tPK7block_t (fish)
                                                                 #46 0x000055be89d152fa _ZN25parse_execution_context_t16run_if_statementERK12parse_node_t (fish)
                                                                 #47 0x000055be89d1442b _ZN25parse_execution_context_t9run_1_jobERK12parse_node_tPK7block_t (fish)
                                                                 #48 0x000055be89d1473e _ZN25parse_execution_context_t12run_job_listERK12parse_node_tPK7block_t (fish)
                                                                 #49 0x000055be89d15d60 _ZN25parse_execution_context_t19eval_node_at_offsetEjPK7block_tRK10io_chain_t (fish)
                                                                 #50 0x000055be89d221e2 _ZN8parser_t15eval_block_nodeEjRK10io_chain_t12block_type_t (fish)
                                                                 #51 0x000055be89d2249f _ZN8parser_t4evalERKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEERK10io_chain_t12block_type_t17parse_node_tree_t (fish)
                                                                 #52 0x000055be89d23ddd _ZN8parser_t4evalERKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEERK10io_chain_t12block_type_t (fish)
                                                                 #53 0x000055be89cdac8f _ZL20internal_exec_helperR8parser_tRKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEj12block_type_tRK10io_chain_t.constprop.157 (fish)
                                                                 #54 0x000055be89cdc49d _Z8exec_jobR8parser_tP5job_t (fish)
                                                                 #55 0x000055be89d1413e _ZN25parse_execution_context_t9run_1_jobERK12parse_node_tPK7block_t (fish)
                                                                 #56 0x000055be89d1473e _ZN25parse_execution_context_t12run_job_listERK12parse_node_tPK7block_t (fish)
                                                                 #57 0x000055be89d15d60 _ZN25parse_execution_context_t19eval_node_at_offsetEjPK7block_tRK10io_chain_t (fish)
                                                                 #58 0x000055be89d221e2 _ZN8parser_t15eval_block_nodeEjRK10io_chain_t12block_type_t (fish)
                                                                 #59 0x000055be89d2249f _ZN8parser_t4evalERKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEERK10io_chain_t12block_type_t17parse_node_tree_t (fish)
                                                                 #60 0x000055be89d23ddd _ZN8parser_t4evalERKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEERK10io_chain_t12block_type_t (fish)
                                                                 #61 0x000055be89cd86fb _ZL19event_fire_internalRK7event_t (fish)
                                                                 #62 0x000055be89cd9460 _Z10event_firePK7event_t (fish)
                                                                 #63 0x000055be89cd9626 _Z18event_fire_genericPKwPSt6vectorINSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEESaIS7_EE (fish)
-- Subject: 进程 31751 (fish) 核心已转储
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- Documentation: man:core(5)
-- 
-- 进程 31751 (fish) 已崩溃并进行核心转储。
-- 
-- 这通常意味着崩溃程序中存在编程错误，并应当将此错误向其开发者报告。
12月 11 14:58:56 localhost.localdomain audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@2-32053-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
12月 11 14:58:57 localhost.localdomain abrt-server[32067]: Deleting problem directory ccpp-2018-12-11-14:58:56.528181-31751 (dup of ccpp-2018-12-11-14:50:27.1514-28682)
12月 11 14:58:57 localhost.localdomain org.freedesktop.problems[875]: abrt-dbus: Lock file '.lock' is locked by process 32113
12月 11 14:58:57 localhost.localdomain abrt-notification[32114]: Process 28682 (fish) crashed in append_formatv(std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >&, wchar_t const*, __va_list_tag*)()
-- Subject: ABRT has detected unexpected termination: fish
-- Defined-By: ABRT
-- Support: https://bugzilla.redhat.com/
-- Documentation: man:abrt(1)
```

so what can i do to fix it?

**Comment by zanchey** [comment]
This is probably a duplicate of #4847, which is fixed in the upcoming fish 3.0.

If you disable your localisation (eg run fish with `env LC_ALL=C fish`), this problem should go away.

**Comment by zd5043039119** [comment]
Cool, it really solved. Thx!
